### DISCUSSION TITLE
Expected variants in ZC4H2 not found by seqr

### DISCUSSION THREAD

**Discussion by jxchong** [broadinstitute/seqr#2320] [Q&A] [answered]
We have a project/VCF (hg19) that contains multiple unrelated individuals and we know that 4 of the subjects each have one causal variant in ZC4H2. No matter what mode of inheritance I use / what filter parameters, I can't get these variants to show up except if I do a search by coordinates and provide the gene's coordinates. Then, some of the variants show up in the results but are not listed as being in the gene (no annotations/impact). Presumably the lack of impact/annotation to ZC4H2 is the immediate cause of why they're not showing up in regular variant search but I'm not sure why they're not annotated correctly. 

The variants are:
X:64138986 TG>T  frameshift
X:64139056 G>A stop_gain/nonsense
X:64139085 C>A splice_acceptor
X:64139960 C>T splice_donor

I'm attaching a VCF slice for this gene. The VCF contains annotations generated by VEP through our non-seqr pipeline and you can see that VEP annotates the consequences and gene correctly.

**Comment by hanars** [comment, marked as answer]
Theres not a lot of detail here for me to work off of, but some things to consider are:
- Are you sure you have loaded data to seqr from the VCF? You are seeing data loaded for all the individuals you expect in seqr?
- Are you sure the variants you are looking for are in the VCF you loaded itself?
- Are you searching across all the families? seqr is primarily designed to search for variants within one family at a time and it sounds like these variants are in unrelated individuals so you need to make sure all of their families are included in the search
- What search criteria are you actually  using? As a sanity check, do a search with no filters applied except for the gene name. This means no inheritance filters whatsoever, no frequency, pathogenicity, annotation etc. Do not select one of the saved searches and then clear fields, just add the gene name and hit search.

**Reply by jxchong** [reply]
For your first 3 questions - yes because as I mentioned, I can retrieve the variants in question if I search seqr by coordinates but NOT if I do a regular inheritance search, regardless of whether it's subset by the gene name or if I'm searching the entire project.

Question 4 - search criteria. Yes, I did do a sanity check (screenshot) - a search with absolutely no filters except the gene name and got only 2 intronic common variants in the gene. These two variants are included in the VCF slice I'm attaching to this post (sorry the one above was the wrong slice). The VCF slice contains annotations from running VEP v89.

I suspect the reason the variants aren't coming up is because the seqr loading process is incorrectly annotating them as being not in ZC4H2 - actually it's annotating them as not being in any gene at all. 


![Screen Shot 2021-11-23 at 9 35 50 AM](https://user-images.githubusercontent.com/1133838/143075981-0aed9898-1585-482a-8058-51ba085a1f9b.png)
l
![Screen Shot 2021-11-23 at 9 40 04 AM](https://user-images.githubusercontent.com/1133838/143076201-aec513ce-a28b-431f-b672-bfa6d1dd5864.png)

[zc4h2.vcf.gz](https://github.com/broadinstitute/seqr/files/7590055/zc4h2.vcf.gz)

.

**Reply by hanars** [reply]
Thanks, that is really helpful! You are correct that it looks like variants are not annotated as being in any transcripts - this indicates to me that VEP was not properly run as part of the loading pipeline. One way this might happen is if you run VEP on your joint called VCF before trying to load it with the pipeline - if that is the case, try processing your VCF to drop all VEP annotations, and then run it through the seqr loading pipeline again. If thats not the case, maybe there is an issue with the VEP configuration you are using in the loading pipeline - could you send me the logs from the data loading job?

**Reply by karynne7** [reply]
What I suspect is going on, is that this region doesn't liftover between 37 and 38 (at least when I tried going back to 38 on that slice, it said no target). After vep annotates in 38 and lifts over, (not 100% on the process) your annotations aren't correct in 37 in this case. Let us know if we can provide further details. Thanks!

**Reply by hanars** [reply]
So like I said, you are attempting to load an input VCF with VEP already run on it. You can see that from the line in the logs that states:
```
MT using schema class SeqrVariantsAndGenotypesSchema already has "vep" annotation.
```

You need to make sure there is not vep annotation on your VCF before you load it

**Reply by karynne7** [reply]
Interestingly, we have not run any of the VCFs I've uploaded through VEP and the logs all have that line, even grc38 VCF logs. Not sure what's triggering that line, any thoughts on GATK annotations we should remove? There must be something it's looking for.

**Reply by jxchong** [reply]
Another thing that might help is that other variants in the project *do* have gene/impact annotations assigned to them, so I assume seqr's VEP is running successfully for those variants.

![Screen Shot 2021-11-23 at 12 32 25 PM](https://user-images.githubusercontent.com/1133838/143103094-18892812-f191-46ef-b047-ce6b691806ee.png)

**Reply by hanars** [reply]
so those variants/ genes you are showing are not "seqr's VEP", that is the VEP coming in already annotated on the VCF. If the pipeline sees that VEP has already been run on any of the variants it will not run VEP at all. It seems to me that however you generated or processed the VCF before loading it to seqr, you somehow managed to call VEP on it but not in a way that properly called that gene.  As I suggested above, try processing your VCF to drop all VEP annotations (the annotation called "vep"), and then run it through the seqr loading pipeline again. I would also make sure to delete any intermediary MTs so the pipeline doesn't reuse the malformed data at all

**Reply by jxchong** [reply]
![Screen Shot 2021-11-23 at 12 50 04 PM](https://user-images.githubusercontent.com/1133838/143114361-9ef5892e-d584-4e3d-8a40-fe132e5dbaa7.png)

This is a screenshot of the VCF we uploaded. There's no `CSQ=` in the INFO field. I also get no results when I run `tabix -H bamshad_uwcmg_csmid_1.HF.final.vcf.gz.VT.vcf.gz | grep vep`

**Reply by hanars** [reply]
Well the matrix table that is loaded from that VCF has "vep" already according to the logs you sent me. I suppose it is possible thats not the underlying issue, but that seems like a bad sign to me. You should have an intermediate matrix table that was generated by the pipeline - maybe check and see if for those particular variants you are seeing vep annotating properly or not in the Matrix Table, so we can isolate where in the process this went wrong?

**Reply by jxchong** [reply]
What tags/fields in the VCF trigger that error message (about having "vep" already)?

**Reply by hanars** [reply]
I'm sorry, digging in to this more it looks like the way we run our pipeline is quite different from the way we have local installations run their pipelines so that may be an expected warning for local installations. It would be helpful if in addition to the logs you could also send the actual command you ran. I also would still like you to check the intermediate MT generated by the pipeline to see what the vep annotations are for the variants of interest

**Reply by jxchong** [reply]
I'll let @karynne7 respond since she does that part, but FYI we're (UW) actually running in Google Cloud, not locally as we thought it might be easier to keep in sync.

**Reply by karynne7** [reply]
Looking into the MT, but for now this was my command

**Reply by karynne7** [reply]
[hailmt.txt](https://github.com/broadinstitute/seqr/files/7591647/hailmt.txt)
I'm not the quickest with Hail, but I subset the mt to just gene_symbol.contains('ZC4H2') and only got 2 variants. This occurred using vep/, mainTranscripts/, and sortedTranscriptConsequences/. If there is another field I should look at, let me know.
Then, I filtered by the region (X:64135682-64196413) and got back 7 variants we'd expect.

**Reply by hanars** [reply]
so that confirms that this is a pipeline issue, which is good to know. For the 7 variants that come back in the reqion, can you confirm what the "vep" annotation in the matrix table is? It would be helpful to know if they were not being called by VEP at all or if VEP was calling them but not within the expected gene. 

In terms of how VEP is being called, our instructions for running the pipeline include a `  --vep-config-json-path` argument which you seem to be omitting. You are therefore using whatever the default vep configuration would be in a hail dataproc cluster. While this should be fine, as hail defaults are generally good ones, that might be where you are running into issues. If hail is not annotating VEP on these variants using the default VEP configuration that may actually be a question to ask the hail team about and not our team.

**Reply by karynne7** [reply]
So I added the argument 
```--vep-config-json-path "gs://uw-mgrc/vep85-loftee-gcloud.json" ```
(from gs://hail-us-vep/vep85-loftee-gcloud.json, you pointed me to) and recovered the missing annotation. I am using the cloud and option #1 from your guide, which doesn't specify using that argument. (https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md) 
I know you've told me to use it before, but the file I previously used VEP-GRCh37.json and always got errors. (https://github.com/broadinstitute/seqr/tree/master/deploy/docker/pipeline-runner/VEP-GRCh37.json) 
I'm glad it's working, but I'm not sure about *why* the differences are there. I also wanted to mention, this doesn't fix the tolerate_parse_error I got before, I still require that fix for grch37. Thanks, as always, for the help Hana. 

```
diff VEP-GRCh37.json vep85-loftee-gcloud.json 
11,12c11
<     "--dir_cache", "/vep_data",
<     "--plugin", "LoF,human_ancestor_fa:/vep_data/loftee_data/GRCh37/loftee_data/human_ancestor.fa.gz,filter_position:0.05,min_intron_size:15,conservation_file:/vep_data/loftee_data/GRCh37/loftee_data/phylocsf_gerp.sql,gerp_file:/vep_data/loftee_data/GRCh37/loftee_data/GERP_scores.final.sorted.txt.gz",
---
>     "--plugin", "LoF,human_ancestor_fa:/root/.vep/loftee_data/human_ancestor.fa.gz,filter_position:0.05,min_intron_size:15,conservation_file:/root/.vep/loftee_data/phylocsf_gerp.sql,gerp_file:/root/.vep/loftee_data/GERP_scores.final.sorted.txt.gz",
16c15
<      "PERL5LIB": ""
---
>      "PERL5LIB": "/vep_data/loftee"
19c18
< }
---
> }
```

**Reply by hanars** [reply]
I'm glad adding the VEP config fixed it! This is an oversight in our documentation so I will make sure we update it, thank you for letting me know
